apache/trafficcontrol
# Improve tc package GoDoc

## What does this PR (Pull Request) do?

This pull request improves the GoDoc for the `tc` package by fixing 110 comment-related warnings generated by Golint (before this PR, there were 273 comment-related warnings; now it is down to 163 comment-related warnings).
- [x] This PR addresses some of #3288, though that is a very large issue and closing it is beyond the scope of this PR. 

This pull request also improves documentation for `enum.go` by removing the part of its overview that refers to it as a package, since we want to keep it part of the `tc` package (see https://github.com/apache/trafficcontrol/pull/4293#issuecomment-574356841 and 3f3d778a1b)

- [x] This PR fixes #4295 

Because only comments are modified or added and no actual code, no tests are included in this PR.

## Which Traffic Control components are affected by this PR?


- Documentation

## What is the best way to verify this PR?

* Although there are still Golint warnings, no warnings should be issued for the comments changed or added. Check comment-related warnings with:
```shell script
  golint lib/go-tc | grep comment
  ```

* Check the GoDoc:
    - ```shell script
      godoc -http=:6060
      ```
    - Visit http://localhost:6060/pkg/github.com/apache/trafficcontrol/lib/go-tc in a browser

## If this is a bug fix, what versions of Traffic Control are affected?


- master (6677a9c07)
- 3.1.0
- 4.0.0 (RC1)

## The following criteria are ALL met by this PR




## Additional Information
